### DISCUSSION TITLE
Testing pimcore

### DISCUSSION THREAD

**Discussion by msoroka** [pimcore/pimcore#11051] [Q&A / Support]
Hello, 

I want to write some tests to my application in Codeception. Documentation says that I can use Pimcore's core helpers but.. I can't actually. 

In pimcore/pimcore repository in [gitattributes](https://github.com/pimcore/pimcore/blob/10.x/.gitattributes) /tests directory is ignored so when creating new project or cloning new repository with pimcore there is no tests catalog in namespace mentioned in [documentation](https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Testing/Application_Testing.html). To be precised it's about these line
`require_once PIMCORE_PROJECT_ROOT . '/vendor/pimcore/pimcore/tests/_support/Util/Autoloader.php';`

I want to understand, the documentation is outdated or there is mistake in pimcore repository that tests directory is ignored or maybe I do something wrong? Does anybody know any other methods for testing pimcore?

**Comment by jdreesen** [comment]
You have to install pimcore by source (`prefer-source`), see: #10226

**Reply by jdreesen** [reply]
Maybe you can create a PR with updated docs for this?

**Reply by msoroka** [reply]
I followed the steps you mentioned in issue and now I have tests directory in proper namespace. But it's workaround in my opinion and it should be fixed by other way, for example by removing "tests" from excluded directories from git.
